src/main/java/com/mineboundteam/minebound/magic/PassiveSpellItem.java

```diff
@@ -65,11 +65,22 @@ protected static <T extends PassiveSpellItem> ItemStack getHighestEquippedSpellO
         return highestSpell;
     }
 
-    protected static <T extends PassiveSpellItem> T getHighestSpellItem(Class<T> type, Player player) {
-        return getHighestSpellItem(getEquippedSpellItemsOfType(type, player));
+    @SuppressWarnings("unchecked")
+    public static <T extends PassiveSpellItem> T getHighestSpellItem(Class<T> type, Player player) {
+        T highestSpell = null;
+        for (EquipmentSlot e : EquipmentSlot.values())
+            if(e.getType() == EquipmentSlot.Type.ARMOR && player.getItemBySlot(e).getItem() instanceof MyrialArmorItem)
+                for (Tag tag : ArmorNBTHelper.getSpellTag(player.getItemBySlot(e), ArmorNBTHelper.PASSIVE_SPELL)) {
+                    if (tag instanceof CompoundTag cTag) {
+                        ItemStack item = ItemStack.of(cTag);
+                        if (type.isInstance(item.getItem()) && ((T)item.getItem()).level.getValue() > highestSpell.level.getValue())
```

Comment:
`highestSpell` is `null` the first time around and will crash the game. Will need to do something similar to `getHighestSpellItem` below after checking `isInstance`.

Reply:
fixed